src/vs/workbench/contrib/preferences/browser/settingsWidgets.ts

```diff
@@ -563,14 +578,93 @@ export class ListSettingWidget extends AbstractListSettingWidget<IListDataItem>
 		] as IAction[];
 	}
 
-	protected renderItem(item: IListDataItem): HTMLElement {
+	private dragDetails: ListSettingWidgetDragDetails | undefined;
+
+	private getDragImage(item: IListDataItem): HTMLElement {
+		const dragImage = $('.monaco-drag-image');
+		dragImage.textContent = item.value.data;
+		return dragImage;
+	}
+
+	protected renderItem(item: IListDataItem, idx: number): HTMLElement {
 		const rowElement = $('.setting-list-row');
 		const valueElement = DOM.append(rowElement, $('.setting-list-value'));
 		const siblingElement = DOM.append(rowElement, $('.setting-list-sibling'));
 
 		valueElement.textContent = item.value.data.toString();
 		siblingElement.textContent = item.sibling ? `when: ${item.sibling}` : null;
 
+		if (this.inReadMode) {
+			rowElement.draggable = true;
+			rowElement.classList.add('draggable');
+		} else {
+			rowElement.draggable = false;
+			rowElement.classList.remove('draggable');
+		}
+
+		this._register(DOM.addDisposableListener(rowElement, DOM.EventType.DRAG_START, (ev) => {
```

Comment:
I think these should be on `this.listDisposables` instead so they are properly disposed when rerendering